### ISSUE TITLE
consul not deregistering services automatically when service goes down

### ISSUE THREAD

**Issue by thangavaishnavi** [spring-cloud/spring-cloud-consul#144]
Here is my yaml configuration
```
spring:    
    cloud:
       consul:
          config:
             enabled: false
             uri: http://10.108.54.38/security-web
          discovery:
             ipAddress: http://10.108.54.38/security-web
             healthCheckPath: ${management.contextPath}
             healthCheckInterval: 15s
             address: ${management.contextPath}
    endpoints:
          shutdown:
             enabled: true
    application:
       name: security-web
       address: http://10.108.54.38/security-web
spring.cloud.consul.config.enabled: false
spring.cloud.consul.discovery.heartbeat.enabled: true
management:
    contextPath: /security-web/poll
```

**Comment by spencergibb** [comment]
Can you explain a bit more? I can verify that if I `kill -9` a service registered to consul, that registration doesn't go away. It's status moves to failing. I'm afraid there's nothing I can do about that. If you also mean, that the failing registration is still included in ribbon, then that is a duplicate of #116. If you issue is something else please feel free to clarify and I'll take it into advisement.

**Comment by thangavaishnavi** [comment]
Yes, I meant my services shows failing in consul once i down the service. I am doing service registration using annotatio @EnableDiscoveryclient.

Once is define this annotation in my boot main class, it will take all needed things from yml file.

Now my question how I can make my service to deregister by keeping some value in yml property file??

**Comment by spencergibb** [comment]
How are you stopping the service?

**Comment by thangavaishnavi** [comment]
I will start n stop through command prompt in windows

**Comment by spencergibb** [comment]
As long as the spring application context shuts down properly, it should be deregistered. Not sure if Ctrl-C on the windows cmd prompt does that. It does on a mac/linux.

**Comment by thangavaishnavi** [comment]
ok thank you very much. I will check on that.

**Comment by vtahlani** [comment]
It worked for me in Unix and service got de-register from consul when I used "Ctrl+C" to bring down the service.

However service doesn't get de-register if  I click "**Terminate** button in Eclipse console", so i achieved the same by following below then service got de-registered from consul.

1. Added spring cloud actuator dependency , which adds all endpoints health, shutdown etc
2. Used API - localhost:XYZO/shutdown    // here XYZO is placeholder for port on which service is running.

output 
o.s.c.consul.discovery.ConsulLifecycle   : Deregistering service with consul: greeting

don't see service anymore in consul
localhost:8500/v1/catalog/services
